crates/astria-conductor/src/conductor.rs

```diff
@@ -109,22 +106,21 @@ impl Conductor {
 
             let executor = Executor::builder()
                 .rollup_address(&cfg.execution_rpc_url)
-                .rollup_id(rollup_id)
-                .sequencer_height_with_first_rollup_block(cfg.initial_sequencer_block_height)
                 .block_channel(block_rx)
                 .shutdown(shutdown_rx)
                 .set_optimism_hook(hook)
                 .build()
                 .await
                 .wrap_err("failed to construct executor")?;
+            let genesis = executor.get_genesis();
             let executable_sequencer_block_height = executor
                 .calculate_executable_block_height()
                 .wrap_err("failed calculating the next executable block height")?;
 
             tasks.spawn(Self::EXECUTOR, executor.run_until_stopped());
             shutdown_channels.insert(Self::EXECUTOR, shutdown_tx);
 
-            (block_tx, executable_sequencer_block_height)
+            (block_tx, genesis, executable_sequencer_block_height)
```

Comment:
side remark: i am switching around the logic in my refactor so that this block just returns the executor and the other tasks can be spun up by using e.g. `celestia_namespace_v0_from_rollup_id(executor.genesis_info().rollup_id())` because the executor task is launched later.